### ISSUE TITLE
Deployment failure due to "lack of deployer pod" is ambiguous

### ISSUE THREAD

**Issue by kargakis** [openshift/origin#4942]
When a deployment is in pending or running state and the deployer pod is nowhere to be found (https://github.com/openshift/origin/blob/8ba29417051227f61f7f641ef04ade9fe7f48dc9/pkg/deploy/controller/deployment/controller.go#L120), we report that the deployment failed because the deployer pod does not exist. Though that's the case with successful deployment too. I believe we should fix the wording for [this](https://github.com/openshift/origin/blob/ac969ccd3580b6d18392002df541d64bac7ac471/pkg/deploy/api/types.go#L221). 

cc: @ironcladlou

**Comment by ironcladlou** [comment]
I suppose the fundamental problem is we're not actually capable of detecting whether the deployment state matches the user intent without relying on the outcome of the deployer pod. So the state really is "unknown." The new upstream deployment architecture addresses this because the system can use state observation to know whether the deployment is "done" without relying on watching a coordinating/volatile process.

If the deployer pod succeeded but was deleted before we had a chance to inspect it and update the deployment status, we assume failure since we just don't know for sure given the architecture.

Is there any evidence of deployer pods frequently being reaped before we can update the deployment status? If not, this might be very low priority given our intent to move to the upstream architecture.

**Comment by kargakis** [comment]
> Is there any evidence of deployer pods frequently being reaped before we can update the deployment status? If not, this might be very low priority given our intent to move to the upstream architecture.

I don't think so, I hit it while working on https://github.com/openshift/origin/pull/4941 and was confused by the message, though if it is already solved upstream then I guess this is indeed very low priority.

**Comment by kargakis** [comment]
Fixed by https://github.com/openshift/origin/pull/4963
